tests/metrics_test.go

```diff
@@ -153,14 +158,95 @@ var _ = Describe("Metrics", func() {
 		)
 	})
 
+	Context("SSP metrics", func() {
+		getDeployment := func(name string, namespace string) *apps.Deployment {
+			deployment := &apps.Deployment{}
+			err := apiClient.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, deployment)
+			Expect(err).ToNot(HaveOccurred())
+			return deployment
+		}
+
+		getOperatorPod := func() (k8sv1.Pod, uint16) {
+			pods, port := operatorPodsWithMetricsPort()
+			return pods[0], port
+		}
+
+		updateVersion := func(version string) {
+			deployment := getDeployment(sspDeploymentName, sspDeploymentNamespace)
+
+			envs := deployment.Spec.Template.Spec.Containers[0].Env
+			var newEnvs []k8sv1.EnvVar
+
+			for _, env := range envs {
+				if env.Name != common.OperatorVersionKey {
+					newEnvs = append(newEnvs, env)
+				}
+			}
+
+			newEnvs = append(newEnvs, k8sv1.EnvVar{Name: common.OperatorVersionKey, Value: version})
+			deployment.Spec.Template.Spec.Containers[0].Env = newEnvs
+
+			err := apiClient.Update(ctx, deployment)
```

Comment:
This update may fail, because of a concurrent update operation. Can you wrap the update logic in an `Eventually()` ?